src/faultlog/deconfig_records.cpp

```diff
@@ -0,0 +1,130 @@
+#include <attributes_info.H>
+#include <fmt/format.h>
+
+#include <deconfig_reason.hpp>
+#include <deconfig_records.hpp>
+#include <phosphor-logging/log.hpp>
+
+extern "C"
+{
+#include <libpdbg.h>
+}
+namespace openpower::faultlog
+{
+
+using ::nlohmann::json;
+using ::phosphor::logging::level;
+using ::phosphor::logging::log;
+
+constexpr auto STATE_CONFIGURED = "CONFIGURED";
+constexpr auto STATE_DECONFIGURED = "DECONFIGURED";
+
+struct DeconfigDataList
+{
+    std::vector<pdbg_target*> targetList;
+    void addPdbgTarget(pdbg_target* tgt)
+    {
+        targetList.push_back(tgt);
+    }
+};
+
+/**
+ * @brief Get pedgt targets which has been deconfigured
+ *
+ * This callback function is called as part of the recursive method
+ * pdbg_target_traverse, recurtion will exit when the method return 1
+ * else continues till the target is found
+ *
+ * @param[in] target - pdbg target to compare
+ * @param[inout] priv - data structure to pass to the callback method
+ *
+ * @return 1 when target is found else 0
+ */
+static int getDeconfigTargets(struct pdbg_target* target, void* priv)
+{
+    auto deconfigList = reinterpret_cast<DeconfigDataList*>(priv);
+    ATTR_HWAS_STATE_Type hwasState;
+    if (!DT_GET_PROP(ATTR_HWAS_STATE, target, hwasState))
+    {
+        if ((hwasState.deconfiguredByEid != 0) &&
+            (DECONFIGURED_BY_PLID_MASK & hwasState.deconfiguredByEid) == 0)
+        {
+            // inlcude only specific states and other might be by association
+            if ((hwasState.deconfiguredByEid == DECONFIGURED_BY_MANUAL_GARD) ||
```

Comment:
To make it more readable and easier to extend in the future, can we use 

switch (hwasState.deconfiguredByEid) {
    case DECONFIGURED_BY_MANUAL_GARD:
    case DECONFIGURED_BY_FIELD_CORE_OVERRIDE:
    case DECONFIGURED_BY_PRD:
    case DECONFIGURED_BY_PHYP:
    case DECONFIGURED_BY_SPCN:
        deconfigList->addPdbgTarget(target);
        break;
    default:
        // handle other cases
        break;
}

or a vector containing the elements.  

#include <vector>

std::vector<num> reasonsToCheck = {DECONFIGURED_BY_MANUAL_GARD,
                                   DECONFIGURED_BY_FIELD_CORE_OVERRIDE,
                                   DECONFIGURED_BY_PRD,
                                   DECONFIGURED_BY_PHYP,
                                   DECONFIGURED_BY_SPCN};

if (std::find(reasonsToCheck.begin(), reasonsToCheck.end(),
              hwasState.deconfiguredByEid) != reasonsToCheck.end())
{
    deconfigList->addPdbgTarget(target);
}